WooCommerce/Classes/ViewRelated/Dashboard/Settings/Privacy/PrivacySettingsViewController.swift

```diff
@@ -445,6 +448,42 @@ private extension PrivacySettingsViewController {
     }
 }
 
+// MARK: - Notices
+//
+private extension PrivacySettingsViewController {
+    /// Presents an error notice when failing to fetch the account settings.
+    /// The retry button calls the `pullToRefresh` method.
+    ///
+    func presentErrorFetchingAccountSettingsNotice() {
+        // Needed to treat every notice as unique. When not unique the notice presenter won't display subsequent error notices.
+        let info = NoticeNotificationInfo(identifier: UUID().uuidString)
+        let notice = Notice(title: Localization.errorFetchingAnalyticsState,
+                            feedbackType: .error,
+                            notificationInfo: info,
+                            actionTitle: Localization.retry) { [weak self] in
+            guard let self else { return }
+            self.pullToRefresh(sender: self.refreshControl)
+        }
+        ServiceLocator.noticePresenter.enqueue(notice: notice)
+    }
+
+    /// Presents an error notice when failing to update the account settings.
+    /// Receives the intended analytics `optInValue`as a parameter to be able to resubmit the request upon a retry.
+    ///
+    func presentErrorUpdatingAccountSettingsNotice(optInValue: Bool) {
+        // Needed to treat every notice as unique. When not unique the notice presenter won't display subsequent error notices.
+        let info = NoticeNotificationInfo(identifier: UUID().uuidString)
+        let notice = Notice(title: Localization.errorUpdatingAnalyticsState,
+                            feedbackType: .error,
+                            notificationInfo: info,
+                            actionTitle: Localization.retry) { [weak self] in
+            guard let self else { return }
+            self.collectInfoWasUpdated(newValue: optInValue)
```

Comment:
Because we have `self?.collectInfo = !newValue` when the initial action fails, should we set `self?.collectInfo = optInValue` in this retry action?

Otherwise, when you tap "Retry" it can update successfully but the change isn't reflected in the UI until you refresh.

Reply:
Nice catch!